src/rules/valid-expect.ts

```diff
@@ -90,6 +90,11 @@ const isAcceptableReturnNode = (
     AST_NODE_TYPES.AwaitExpression,
   ].includes(node.type);
 
+const isNoAssertionsParentNode = (node: TSESTree.Node): boolean =>
+  node.type === AST_NODE_TYPES.ExpressionStatement ||
+  (node.type === AST_NODE_TYPES.AwaitExpression &&
+    node.parent!.type === AST_NODE_TYPES.ExpressionStatement);
```

Comment:
It's generally best to avoid using the non-null assertion operator where possible, as it reduces the effectiveness of TypeScript, and it is technically possible for `node.parent` to be undefined.

Instead just using `node.parent &&` should be enough :)

Reply:
`node.parent &&` doesn't typecheck because it's not strictly a boolean. I'll go with `node.parent !== undefined &&` if that's OK with you